src/go/rpk/pkg/cli/cmd/redpanda/admin/partitions/partitions.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// Package partitions contains commands to talk to the Redpanda's admin partitions
+// endpoints.
+package partitions
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+)
+
+// NewCommand returns the partitions admin command.
+func NewCommand(fs afero.Fs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "partitions",
+		Short: "View and configure Redpanda partitions through the admin listener.",
+		Args:  cobra.ExactArgs(0),
+	}
+	cmd.AddCommand(
+		newListCommand(fs),
+	)
+	return cmd
+}
+
+func newListCommand(fs afero.Fs) *cobra.Command {
+	var (
+		leaderOnly bool
+	)
+	cmd := &cobra.Command{
+		Use:     "list [BROKER ID]",
+		Aliases: []string{"ls"},
+		Short:   "List the partitions in a broker in the cluster.",
+		Args:    cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			brokerId, err := strconv.Atoi(args[0])
+			out.MaybeDie(err, "invalid broker %s: %v", args[0], err)
+			if brokerId < 0 {
+				out.Die("invalid negative broker id %v", brokerId)
+			}
+
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			adm, err := kafka.NewAdmin(fs, p, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer adm.Close()
+
+			var m kadm.Metadata
+			m, err = adm.Metadata(context.Background())
```

Comment:
How is this command different from `rpk cluster metadata -d`? That command prints the metadata as well per topics and their leader.

Reply:
`rpk cluster metadata -d` prints all topics, partitions with leader and replica information from cluster's perspective. This makes complete sense as we are using `rpk cluster .... ` command.

Now, if we want to see partition information from a broker's perspective then we need another command.
`rpk redpanda admin partitions list [BROKER ID]`

Reply:
Also, in the future we can have partition level commands such as:

```
rpk redpanda admin partitions drain [BROKER ID]
rpk redpanda admin partitions move [ SOURCE BROKER ID ] [ DESTINATION BROKER ID ]
rpk redpanda admin partitions something else ....
```

Reply:
@twmb - Let me know if this all makes sense or not. Thanks!

Reply:
`rpk cluster metadata` accepts topics as well to filter which topics to query metadata for. I see the point for wanting to know the partitions that live on a specific broker, though; that's a nice feature. I wonder if this would better be suited for the admin interface rather than a kafka protocol request?

w.r.t. drain / move / ls, this is a good batch of features and I'm not sure where that should fit in the API yet. It seems to be that these features would be pretty general for any admin.

Reply:
In my honest opinion, we should optimize in the code whether to use Kafka protocol or Admin API or a combination of the two in the code. For the external user we should make the commands super intuitive. 

Is there anyone else we should include here to review this command's usage ? - @twmb

Reply:
I think since we're accepting that most things under `rpk redpanda admin` should move or be improved in the future, that this can be fine for now -- backwards compatibility may be a bit difficult, but we can live with that, what do you think about this plan?

Reply:
Sounds good to me. I'll resolve this comment then. Thank you so much @twmb for the review!